guacml/step_tree/model_manager.py

```diff
@@ -0,0 +1,55 @@
+from guacml.step_tree.feature_reducer import FeatureReducer
+from guacml.step_tree.hyper_param_optimizer import HyperParameterOptimizer
+from guacml.step_tree.model_result import ModelResult
+
+from guacml.step_tree.model_runner import ModelRunner
+from .base_step import BaseStep
+
+
+class ModelManager(BaseStep):
+    def __init__(self, model, target, hyper_param_iterations, eval_metric):
+        self.model = model
+        self.target = target
+        self.hyper_param_iterations = hyper_param_iterations
+        self.eval_metric = eval_metric
+        self.splitter = None
+
+    def execute(self, input, metadata):
+        model_runner = ModelRunner(self.model, input, self.target, self.eval_metric, self.splitter)
+        features = self.model.select_features(metadata)
+        features = features[features != self.target]
+
+        hp_optimizer = HyperParameterOptimizer(model_runner, features)
+        all_trials, best_hps = hp_optimizer.optimize(self.hyper_param_iterations)
+
+        feature_reducer = FeatureReducer(model_runner, best_hps)
```

Comment:
If we move all this to the tree, then it could also be decided by the user. The `HyperParameterOptimiser` is very expensive to run. It needs to retrain the model many times. That's why I just run in once before feature reduction and once after.